aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -171,12 +164,25 @@ Tensor as_strided_tensorimpl_mps(const Tensor& self, IntArrayRef size,
       } else {
         // Else part takes care of the chaining where multiple view operations
         // were implemented on the same underlying data storage ptr
-        cachedGraph->outputTensor_ = chainViewOperation(cachedGraph->graph(),
-                                      size, stride, storage_offset,
-                                      cachedGraph->outputTensor_, self);
-        cachedGraph->size_ = size;
-        cachedGraph->stride_ = stride;
-        cachedGraph->storage_offset_ = storage_offset;
+        string insert_key = mps::getStridedKey(self, size, stride, storage_offset);
+        MPSCachedGraph *tmpCachedGraph = cache_->CreateCachedGraph(insert_key, ^ MPSCachedGraph * () {
```

Comment:
Is this outside of the autoreleasepool?
Or if there is an outer one, is a new one below needed?

Reply:
The outer one takes care of this code. If you notice the inner one is for the lambda used there.